File: go/docmost/models/update_share_dto.go
package models

// UpdateShareDto from OpenAPI.
type UpdateShareDto struct {
	ShareId string `json:"shareId"`
	PageId *string `json:"pageId,omitempty"`
	IncludeSubPages *bool `json:"includeSubPages,omitempty"`
	SearchIndexing *bool `json:"searchIndexing,omitempty"`
}

File: c#/src/Docmost.Sdk/Models/UpdateShareDto.cs
namespace Docmost.Sdk.Models;

public class UpdateShareDto
{
    [System.Text.Json.Serialization.JsonPropertyName("shareId")]
    public string ShareId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("pageId")]
    public string? PageId { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("includeSubPages")]
    public bool? IncludeSubPages { get; set; }

    [System.Text.Json.Serialization.JsonPropertyName("searchIndexing")]
    public bool? SearchIndexing { get; set; }

}
